### PR TITLE
 Adds Type Hints 

### LINKED ISSUE #11: Add support for typehinting
Due to lack of typehinting, it's very hard for us to code in lavasnek_rs.
Kindly look into it!

### PR DESCRIPTION
**Pull request by CaffeineDuck**
Adds TypeHints to `lavasnek_rs`, `lavasnek_rs.pyi` will automatically included by `Maturin` in the wheel it builds.

Fixes #11 as well.

### CODE REVIEW

File: lavasnek_rs.pyi [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    async def seek_secs(self, guild_id: int, time: int) -> None: ...
+    async def jump_to_time_secs(self, guild_id: int, time: int) -> None: ...
+    async def scrub_secs(self, guild_id: int, time: int) -> None: ...
+    async def seek_millis(self, guild_id: int, time: int) -> None: ...
+    async def jump_to_time_millis(self, guild_id: int, time: int) -> None: ...
+    async def scrub_millis(self, guild_id: int, time: int) -> None: ...
+    async def volume(self, guild_id: int, volume: int) -> None: ...
+    async def equalize_all(self, guild_id: int, bands: list[float]) -> None: ...
+    async def equalize_dynamic(self, guild_id: int, bands) -> None: ...
+    async def equalize_band(self, guild_id: int, band) -> None: ...
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
    async def equalize_dynamic(self, guild_id: int, bands: t.List[Band]) -> None: ...
    async def equalize_band(self, guild_id: int, band: Band) -> None: ...
```

File: lavasnek_rs.pyi [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+class PlaylistInfo:
+    selected_track: t.Optional[int]
+    name: t.Optional[str]
+
+class Track:
+    info: Info
+    track: str
+
+class Tracks:
+    tracks: list[Track]
```
Comment (vicky5124) [inline lavasnek_rs.pyi]: 3.6 support
```suggestion
    tracks: t.List[Track]
```

File: lavasnek_rs.pyi [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    track: Track
+    end_time: int
+
+class Node:
+    volume: int
+    is_on_loops: bool
+    guild: int
+    now_playing: t.Optional[TrackQueue]
+    is_paused: bool
+    queue: list[TrackQueue]
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
    queue: t.List[TrackQueue]
```

File: lavasnek_rs.pyi [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    async def get_data(self): ...
+    async def set_data(self, data: t.Union[t.Any, dict]): ...
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
    async def set_data(self, data: t.Union[t.Any, t.Dict]): ...
```

File: lavasnek_rs.pyi [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    async def equalize_all(self, guild_id: int, bands: list[float]) -> None: ...
+    async def equalize_dynamic(self, guild_id: int, bands) -> None: ...
+    async def equalize_band(self, guild_id: int, band) -> None: ...
+    async def equalize_reset(self, guild_id: int) -> None: ...
+    async def remove_guild_from_loops(self, guild_id: int) -> None: ...
+    async def remove_guild_node(self, guild_id: int) -> None: ...
+    async def get_guild_node(self, guild_id: int) -> t.Optional[Node]: ...
+    async def set_guild_node(self, guild_id: int, node: Node) -> None: ...
+    async def get_guild_gateway_connection_info(
+        self, guild_id
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
        self, guild_id: int
```

File: lavasnek_rs.pyi [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    ) -> t.Optional[ConnectionInfo]: ...
+    async def wait_for_full_connection_info_insert(
+        self, guild_id: int, event_count=10
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
        self, guild_id: int, event_count: int=10
```

File: lavasnek_rs.pyi [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    ) -> ConnectionInfo: ...
+    async def wait_for_connection_info_remove(
+        self, guild_id: int, event_count=10
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
        self, guild_id: int, event_count: int=10
```

File: lavasnek_rs.pyi [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    ) -> None: ...
+    async def raw_handle_event_voice_server_update(
+        self, guild_id: int, endpoint, token
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
        self, guild_id: int, endpoint: str, token: str
```

File: lavasnek_rs.pyi [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+    ) -> None: ...
+    async def raw_handle_event_voice_state_update(
+        self, guild_id, user_id, session_id, channel_id: int
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
        self, guild_id: int, user_id: int, session_id: str, channel_id: int
```

File: lavasnek_rs.pyi [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+    async def pause(self, guild_id: int) -> None: ...
+    async def resume(self, guild_id: int) -> None: ...
+    async def seek_secs(self, guild_id: int, time: int) -> None: ...
+    async def jump_to_time_secs(self, guild_id: int, time: int) -> None: ...
+    async def scrub_secs(self, guild_id: int, time: int) -> None: ...
+    async def seek_millis(self, guild_id: int, time: int) -> None: ...
+    async def jump_to_time_millis(self, guild_id: int, time: int) -> None: ...
+    async def scrub_millis(self, guild_id: int, time: int) -> None: ...
+    async def volume(self, guild_id: int, volume: int) -> None: ...
+    async def equalize_all(self, guild_id: int, bands: list[float]) -> None: ...
```
Comment (vicky5124) [inline lavasnek_rs.pyi]:
```suggestion
    async def equalize_all(self, guild_id: int, bands: t.List[float]) -> None: ...
```

**Review by vicky5124 — CHANGES_REQUESTED** [review]
This library supports 3.6, stubs should work for people using that version.
And added so missing typehints as well.

**Review by vicky5124 — APPROVED** [review]

**Comment by CaffeineDuck** [comment]
> This library supports 3.6, stubs should work for people using that version. And added so missing typehints as well.

Thanks for the suggestions, requested changes are applied and resolved.

**Comment by vicky5124** [comment]
If the checks fail again, you will need to rebase most likely.

File: lavasnek_rs.pyi [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+    def finish_time_secs(self, finish: int) -> PlayBuilder: ...
+    def start_time_millis(self, start: int) -> PlayBuilder: ...
+    def finish_time_millis(self, finish: int) -> PlayBuilder: ...
+
+class Lavalink:
+    async def start_discord_gateway(self, wait_time: int) -> None: ...
+    async def join(self, guild_id: int, channel_id: int) -> ConnectionInfo: ...
+    async def leave(self, guild_id: int) -> None: ...
+    async def create_session(self, connection_info: ConnectionInfo) -> None: ...
+    async def destory(self, guild_id: int) -> None: ...
```
Comment (Jonxslays) [inline lavasnek_rs.pyi]: destroy

Comment (vicky5124) [reply lavasnek_rs.pyi]:
```suggestion
    async def destroy(self, guild_id: int) -> None: ...
```

**Review by vicky5124 — CHANGES_REQUESTED** [review]

**Review by vicky5124 — APPROVED** [review]

**Pull request merged**